include/DataFrame/Internals/DataFrame_get.tcc

```diff
@@ -694,6 +694,17 @@ DataFrame<I, H>::get_view_by_idx (Index2D<IndexType> range)  {
         std::upper_bound (indices_.begin(), indices_.end(), range.end);
     DataFrameView<IndexType>    dfv;
 
+    if ((range.begin < indices_.front()) || (range.end > indices_.back()))
+    {
+        char buffer [512];
+
+        sprintf (buffer,
+                "DataFrame::get_view_by_idx(): ERROR: "
+                "Bad begin, end range: %ld, %ld",
+                range.begin, range.end);
+        throw BadRange (buffer);
+    }
+
```

Comment:
@bplaa-yai 
This assumes that the dataframe is entirely sorted by index, This may not be the case at all times.

Reply:
@hosseinmoein indeed. TBH I'm not sure if requesting a view on an unsorted index makes any sense, but I guess one can have a use case where it does. So, what's your point of view on this ? Should bounds checking be left on the user's side, or assume a sorted index as in other parts of the code ? Or any other alternative ?

Reply:
I dislike building interfaces that make assumptions on the state of the object, especially a state that is not obvious. I would just follow the C++/STL idiom of garbage in/garbage out. So I would say just leave it to the user.